contents/docs/api/queries.mdx

```diff
@@ -0,0 +1,264 @@
+
+- `blocking` (default): Executes synchronously unless fresh results exist in cache
+- `async`: Executes asynchronously unless fresh results exist in cache
+- `force_blocking`: Always executes synchronously
+- `force_async`: Always executes asynchronously
+- `force_cache`: Only returns cached results (never calculates)
+- `lazy_async`: Use extended cache period before asynchronous calculation
+- `async_except_on_cache_miss`: Use cache but execute synchronously on cache miss
+
+> **Tip:** To cancel a running query, send a `DELETE` request to the `/api/projects/:project_id/query/:query_id/` endpoint.
+
+### Query types
+
+The `kind` property in the `query` parameter can be one of the following values. 
+
+- `HogQLQuery`: Queries using [PostHog's version of SQL](/docs/sql).
+- `EventsQuery`: Raw event data retrieval
+- `TrendsQuery`: Time-series trend analysis
+- `FunnelsQuery`: Conversion funnel analysis
+- `RetentionQuery`: User retention analysis
+- `PathsQuery`: User journey path analysis
+
+Beyond `HogQLQuery`, these are mostly used to power PostHog internally and are not useful for you, but you can see the [frontend query schema](https://github.com/PostHog/posthog/blob/master/frontend/src/queries/schema/schema-general.ts) for a complete list and more details.
+
+## Response structure
+
+The response format depends on the query type, but all responses include:
+
+- `results`: The data returned by the query
+- `is_cached` (for cached responses): Indicates the result came from cache
+- `timings` (when available): Performance metrics for the query execution
+
+### Cached responses
+
+API queries are cached by default. You can check if a response is cached by checking the `is_cached` property. Responses also contain cache-related details like:
+
+- `cache_key`: A unique identifier for the cached result
+- `cache_target_age`: The timestamp until which the cached result is considered valid
+- `last_refresh`: When the data was last computed
+- `next_allowed_client_refresh`: The earliest time when a client can request a fresh calculation
+
+### Asynchronous queries
+
+For asynchronous queries (like ones with `refresh: async`), the initial response includes a query status with its completion status, query ID, start time, and more:
+
+```json
+{
+  "query_status": {
+    "id": "2fbd4b19413342a4ad08c307155187bc",
+    "team_id": 123,
+    "complete": false
+  }
+}
+```
+
+You can then poll the status by sending a `GET` request to the `/api/projects/:project_id/query/:query_id/` endpoint.
+
+```bash
+curl \
+  -H "Authorization: Bearer $POSTHOG_PERSONAL_API_KEY" \
+  <ph_app_host>/api/projects/:project_id/query/$QUERY_ID/
+```
+
+## Rate limits
+
+API queries on our free plans are limited to:
+
+- 1 query running concurrently
+- 10,000 rows
+- 1TB read bytes during query processing
+
+
+Our ridiculously cheap plan lifts this to:
+
+- 5 queries running concurrently
```

Comment:
It's 3 in RFC, we also need to mention about charing for the bytes read?

Reply:
Once pricing solidifies, we can update this section, just let me know (or feel free to edit it)